plugins/source/azure/client/helpers.go

```diff
@@ -8,7 +8,7 @@ import (
 // this is used to hook ParseResourceGroup and to have easier codegen
 var debug = false
 
-const resourceIDPatternText = `(?i)subscriptions/(.+)/resourceGroups/(.+)/providers/(.+?)/(.+?)/(.+)`
+const resourceIDPatternText = `(?i)subscriptions/(.+)/ResourceGroups/(.+)/providers/(.+?)/(.+?)/(.+)`
```

Comment:
Why this change? was it broken before? Tests set `debug` so this code doesn't really run without a real fetch.

Also we could remove the `const` and just have the string inside `regexp.MustCompile` below, it's not used anywhere else anyway.

Reply:
Reverted